doc/api/n-api.md

```diff
@@ -2788,11 +2793,15 @@ has a reference count of 0. Typically this reference count would be incremented
 temporarily during async operations that require the instance to remain valid.
 
 Caution: The optional returned reference (if obtained) should be deleted via
-[`napi_delete_reference`][] ONLY in response to the finalize callback invocation.
-(If it is deleted before then, then the finalize callback may never be
-invoked.) Therefore when obtaining a reference a finalize callback is also
+[`napi_delete_reference`][] ONLY in response to the finalize callback
+invocation. (If it is deleted before then, then the finalize callback may never
+be invoked.) Therefore when obtaining a reference a finalize callback is also
 required in order to enable correct proper of the reference.
 
+Note: This API may modify the prototype chain of the wrapper object.
+Afterward, additional manipulation of the wrapper's prototype chain may cause
+`napi_unwrap()` to fail.
```

Comment:
Is there some way we can search up the prototype chain so that unless they remove the wrapper object napi_unwrap can still work ?  They could still mess it up but I think it would be less likely to happen.